### ISSUE TITLE
Tensorflow keras version

### ISSUE THREAD

**Issue by KaiLiCn** [rachellea/hirescam#1]
Hi there,

May I ask do you have the codes for the Tensorflow model? 

Kai

**Comment by rachellea** [comment]
Hi Kai - thanks for your question. At this time I don't have a Tensorflow implementation of HiResCAM. However, if you find a Tensorflow implementation of Grad-CAM, you can modify it to get a HiResCAM implementation by removing the gradient averaging step/calculation of importance weights. You can compare the functions in this repo for Grad-CAM vs HiResCAM to see the implementation difference in terms of how the tensors are manipulated: https://github.com/rachellea/hirescam/blob/master/attn_analysis/cams.py lines 84-116.

**Comment by KaiLiCn** [comment]
Great! Thanks!

**Comment by mkwisniewski** [comment]
@KaiLiCn, [here](https://www.kaggle.com/code/mariuszwisniewski/hirescam-for-class-activation-map-visualization) is my notebook where I implemented the *HiResCAM* in TensorFlow, in case you still need it.

**Comment by KaiLiCn** [comment]
Hi @Nexer8 , really appreciate it!
